### PR TITLE
Fix numerous package dependencies, remove test-typescript-2.0 task

### PR DESCRIPTION
**Pull request by adidahiya**
- Compile everything with TypeScript 2.0.10 (previously 2.0.6)
- Remove the now-defunct typescript-2.0 testing task
- Fix up a number of `package.json` dependencies
  - Use React 15.4 consistently in dev environment
  - Do not use floating version ranges in dev environment
  - Do not declare superfluous dependencies in published packages
- Fix a failing unit test for `Button` that broke because of the React and Enzyme bumps

### CODE REVIEW

File: packages/datetime/package.json [hunk 1]
```diff
@@ -13,9 +13,10 @@
   },
   "devDependencies": {
     "bourbon": "4.2.6",
-    "react": "^15.0.1 || ^0.14",
```
Comment (adidahiya) [inline packages/datetime/package.json]: this was the source of the build failures -- React 15.4 was being used with react-addons-test-utils 15.1. These versions are incompatible with each other. Moral of the story: _don't declare floating version ranges in `devDependencies`.

Comment (giladgray) [inline packages/datetime/package.json]: gasp

**Comment by adidahiya** [comment]
ok sweet! now we have a legitimate unit test failure instead of module loading errors / OOMs.

```
Buttons:
    <Button>
      ✔ renders its contents
      ✔ iconName="style" gets icon class
      ✔ clicking button triggers onClick prop
      ✖ clicking disabled button does not trigger onClick prop
      ✔ elementRef receives reference to HTML element
```

**Comment by llorca** [comment]
🎉

**Comment by giladgray** [comment]
opened #229 with fix for unit test (and a pretty sweet refactor 👢 )

**Comment by giladgray** [comment]
thank you @adidahiya!

File: packages/core/src/components/button/buttons.tsx [hunk 2]
```diff
@@ -79,5 +65,22 @@ export class AnchorButton extends React.Component<React.HTMLProps<HTMLAnchorElem
...
+        { [Classes.DISABLED]: props.disabled },
+        Classes.iconClass(props.iconName),
+        Classes.intentClass(props.intent),
+        props.className
+    );
+}
+
+function maybeRenderRightIcon(iconName: string) {
+    if (iconName == null) {
+        return null;
```
Comment (adidahiya) [inline packages/core/src/components/button/buttons.tsx]: lol this was got in the TSLint v4 upgrade PR

File: packages/table/package.json [hunk 3]
```diff
@@ -22,22 +22,22 @@
     "@blueprintjs/core": "^1.0.1",
     "classnames": "^2.2",
     "es6-shim": "^0.35",
-    "react": "^15.0.1 || ^0.14"
+    "pure-render-decorator": "1.2.1"
```
Comment (giladgray) [inline packages/table/package.json]: @adidahiya why the fixed version on this package in table `dependencies`?
core depends on `^1.1`, shouldn't that be enough?

Comment (adidahiya) [inline packages/table/package.json]: hm you're right, this is a mistake, wanna fix it in the 1.1.0 release PR?

**Pull request merged**